pkgs/by-name/ls/lsp-ai/package.nix

```diff
@@ -0,0 +1,80 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  cmake,
+  openssl,
+  zlib,
+  perl,
+  gitUpdater,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "lsp-ai";
+  version = "0.7.0";
+  src = fetchFromGitHub {
+    owner = "SilasMarvin";
+    repo = "lsp-ai";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-5chIz9uIKmwnZG1HjEjqZqoWi3aTmh51r5YJD21u2OE=";
```

Comment:
```
error: hash mismatch in fixed-output derivation '/nix/store/603ynkpwh0xbdzbl5g2gv6kp3cwxas93-source.drv':
         specified: sha256-5chIz9uIKmwnZG1HjEjqZqoWi3aTmh51r5YJD21u2OE=
            got:    sha256-DwqqZBzLevuRCW6QzGyWdE+JtpW6b3EMDuiWtajv/U4=
error: 1 dependencies of derivation '/nix/store/x7r7r2slg86gj5psr9w1jv300s2lda3y-lsp-ai-0.7.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qsvfk6d5as6avd0zpmsknyci8w019b7i-review-shell.drv' failed to build
```

Reply:
```
nix-prefetch-github SilasMarvin lsp-ai --rev release/v0.7.0 --nix
let
  pkgs = import <nixpkgs> {};
in
  pkgs.fetchFromGitHub {
    owner = "SilasMarvin";
    repo = "lsp-ai";
    rev = "476a13ed2dfec064cc4bc8530ba432f30a9dfe9e";
    hash = "sha256-5chIz9uIKmwnZG1HjEjqZqoWi3aTmh51r5YJD21u2OE=";
  }
  ```
  
  Not sure why you are getting an error. I removed the rev has as requested [here](https://github.com/NixOS/nixpkgs/pull/343184#discussion_r1770742519)
  
  It also builds fine 
  ```
  nix-build -A lsp-ai
/nix/store/ryfgpjvvljl14a0kriml17fylqk7q5k1-lsp-ai-0.7.0
```

What command did you use?

Reply:
```
nix-prefetch-github SilasMarvin lsp-ai --rev refs/tags/v0.7.0 --nix
let
  pkgs = import <nixpkgs> {};
in
  pkgs.fetchFromGitHub {
    owner = "SilasMarvin";
    repo = "lsp-ai";
    rev = "26a3a7008d3559757037826d2108c8c1f6ed30de";
    hash = "sha256-DwqqZBzLevuRCW6QzGyWdE+JtpW6b3EMDuiWtajv/U4=";
  }
  ```
  
  Figured it out. Difference in using `release` vs `refs/tags` Also very weird that the hashes still let my local build run just fine.

Reply:
For fixed input derivations once it's in your store it will just use it 

Try

 nix-build -A lsp-ai.src --check

Reply:
Submitted change, should be good now.